### DISCUSSION TITLE
physics sandbox

### DISCUSSION THREAD

**Discussion by vhollander** [jrouwe/JoltPhysics#774] [Q&A] [answered]
good day!

this is a game that im currently porting to use this physics engine as opposed to bullet: https://store.steampowered.com/app/387990
i have few questions to ask:
in the bullet3 implementation, large objects tend to get more bouncy, which makes cars and players jump around as they progress bodies cause lag (that grows exponentially with the objects size) while rebuilding hulls from a manipulation (i.e.. block placement/removal)
objects and childrens create immense lag when there are too many joints attached closely; they also fly around when being collided occasionally
have the issues listed above been addressed in jolt? if not, are there any potential "patches" to apply in the game code?
also, how can i retrieve a body based on a raycast? so that i get to know what is being hit if there is any.

hope to hear from you, many thanks!

**Comment by jrouwe** [comment, marked as answer]
> have the issues listed above been addressed in jolt? if not, are there any potential "patches" to apply in the game code?

From the description it's very hard for me to say what the exact problems are, so I honestly don't know if Jolt will be better or not. From what I hear people say (and from my own measurements), Jolt is about 2x as fast as Bullet, but it will really depend on your scene. If you have integrated Jolt and you have specific scenarios that don't work well, then send me a video / repro and I can take a look at it.

> also, how can i retrieve a body based on a raycast? so that i get to know what is being hit if there is any.

See this:

https://github.com/jrouwe/JoltPhysics/blob/5d0fb0238499149ccb8d21e53fc955132a9a665f/Samples/SamplesApp.cpp#L1025-L1039

**Reply by vhollander** [reply]
thanks a lot for the response!
